src/main/java/uk/gov/companieshouse/chd/order/api/interceptor/ApiKeyAuthorisationInterceptor.java

```diff
@@ -0,0 +1,33 @@
+package uk.gov.companieshouse.chd.order.api.interceptor;
+
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import uk.gov.companieshouse.api.util.security.AuthorisationUtil;
+import uk.gov.companieshouse.api.util.security.SecurityConstants;
+import uk.gov.companieshouse.logging.Logger;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+import static uk.gov.companieshouse.chd.order.api.logging.LoggingUtils.getLogger;
+
+public class ApiKeyAuthorisationInterceptor extends HandlerInterceptorAdapter {
+
+    private static final Logger LOGGER = getLogger();
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+
+        final String identityType = AuthorisationUtil.getAuthorisedIdentityType(request);
```

Comment:
Do we need to call `AuthorisationUtil.hasInternalUserRole()` somewhere too to ensure the key provided represents a internal client with sufficient privileges?

Reply:
I don't believe so. We are CHS_API_KEY for the authentication. If we were using CHS_INTERNAL_API_KEY then we would have to check for that internal role.

Reply:
OK, great! 👍